test/junit/scala/collection/convert/MapWrapperTest.scala

```diff
@@ -79,4 +80,30 @@ class MapWrapperTest {
     assertTrue(jmap == mapWrapper)
     assertTrue(mapWrapper == jmap)
   }
+
+  // was: induce intermittent failure due to contention, where updater is called more than once
+  @Test def `t12586 updateWith should delegate to compute`: Unit = {
+    @volatile var count = 0
+    val jmap = new jutil.concurrent.ConcurrentHashMap[String, String]()
+    class Loki extends Runnable {
+      @volatile var done = false
+      def run(): Unit = {
+        while (!done) {
+          jmap.put("KEY", "VALUE")
+          //Thread.`yield`()
+        }
+      }
+    }
+    val loki = new Loki
+    val runner = new Thread(loki).tap(_.start)
+    val wrapped = jmap.asScala
+    def updater(old: Option[String]) = { count += 1 ; old.map(_ * 2) }
+    for (i <- 1 to 500) {
```

Comment:
The age-old question where to put collections tests is exacerbated by racy collections tests. These ought to be relegated to a special suite that is run occasionally for regression purposes. This arbitrary count reflects convenience in triggering the negative case.

Reply:
I reduced the magic number just for fun.